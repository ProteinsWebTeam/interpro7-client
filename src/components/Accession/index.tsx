import React from 'react';

import cssBinder from 'styles/cssBinder';
import local from './style.css';

const f = cssBinder(local);

type Props = {
  accession: string | number;
  withTitle?: boolean;
  title?: string;
};

const Accession = ({ withTitle, accession, title }: Props) => {
  return (
    <div>
      {title !== '' && withTitle ? <span> {title || 'Accession'}:</span> : null}
      {title === 'Job ID' ? (
        <span className={f('tag-sqc')}> {accession}</span>
      ) : (
        accession
      )}
    </div>
  );
};

export default Accession;
