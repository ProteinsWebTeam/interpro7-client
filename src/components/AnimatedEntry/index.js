import React, { PureComponent } from 'react';
import T from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

const DEFAULT_ITEM_DELAY = 25;
const DEFAULT_DURATION = 250;

/*:: type Props = {
  element: ?Node | string,
  keyframes: ?Array<Object> | Object,
  delay: ?number,
  itemDelay: ?number,
  duration: ?number,
  animateSelf: boolean,
  lowGraphics: boolean,
  dispatch: function,
} */

class AnimatedEntry extends PureComponent /*:: <Props> */ {
  /*::
    _ref: { current: ?HTMLElement };
    _animations: ?Array<any>;
  */
  static propTypes = {
    element: T.oneOfType([T.string, T.element]),
    keyframes: T.oneOfType([T.array, T.object]),
    delay: T.number,
    itemDelay: T.number,
    duration: T.number,
    animateSelf: T.bool,
    lowGraphics: T.bool.isRequired,
    dispatch: T.func.isRequired,
  };

  static defaultProps = {
    element: 'ul',
    keyframes: { opacity: [0, 1] },
    delay: 0,
    itemDelay: DEFAULT_ITEM_DELAY,
    duration: DEFAULT_DURATION,
    animateSelf: false,
  };

  constructor(props) {
    super(props);

    this._ref = React.createRef();
  }

  componentDidMount() {
    if (
      this.props.lowGraphics ||
      !this._ref.current ||
      !this._ref.current.animate
    )
      return;
    const { keyframes, delay, itemDelay, duration, animateSelf } = this.props;
    const toAnimate = Array.from(
      animateSelf ? [this._ref.current] : this._ref.current.children,
    );
    this._animations = toAnimate.map((element /*: any */, i) =>
      element.animate(keyframes, {
        duration,
        delay: delay + itemDelay * (i + 1),
        easing: 'ease-in-out',
        fill: 'both',
      }),
    );
  }

  componentWillUnmount() {
    for (const animation of this._animations || []) {
      if (animation.playState !== 'finished') animation.cancel();
    }
  }

  render() {
    const {
      element: Element,
      // remove those in next lines from props passed
      keyframes,
      delay,
      itemDelay,
      duration,
      animateSelf,
      lowGraphics,
      dispatch,
      ...props
    } = this.props;
    return <Element {...props} ref={this._ref} />;
  }
}

const mapStateToProps = createSelector(
  state => state.settings.ui.lowGraphics,
  lowGraphics => ({ lowGraphics }),
);

export default connect(mapStateToProps, undefined)(AnimatedEntry);
