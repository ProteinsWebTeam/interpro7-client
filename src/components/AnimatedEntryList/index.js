// @flow
import React, {Component} from 'react';
import T from 'prop-types';

const DEFAULT_ITEM_DELAY = 25;
const DEFAULT_DURATION = 250;

export default class extends Component {
  /* ::
    _node: ?Element
    _animations: ?Array<any>
  */
  static propTypes = {
    element: T.oneOfType([
      T.string,
      T.element,
    ]),
    keyframes: T.oneOfType([
      T.array,
      T.object,
    ]),
    delay: T.number,
    itemDelay: T.number,
    duration: T.number,
  };

  static defaultProps = {
    element: 'ul',
    keyframes: {opacity: [0, 1]},
    delay: 0,
    itemDelay: DEFAULT_ITEM_DELAY,
    duration: DEFAULT_DURATION,
  };

  componentDidMount() {
    if (!(this._node && this._node.animate)) return;
    const {keyframes, delay, itemDelay, duration} = this.props;
    this._animations = Array.from(
      this._node.children
    ).map((child/*: any */, i) => child.animate(
      {keyframes},
      {
        duration,
        delay: delay + itemDelay * (i + 1),
        easing: 'ease-in-out',
        fill: 'both',
      },
    ));
  }

  componentWillUnmount() {
    for (const animation of this._animations || []) {
      if (animation.playState !== 'finished') animation.cancel();
    }
  }

  render() {
    const {
      element: Element,
      keyframes, delay, itemDelay, duration, // remove those from props passed
      ...props
    } = this.props;
    return <Element ref={node => this._node = node} {...props} />;
  }
}
