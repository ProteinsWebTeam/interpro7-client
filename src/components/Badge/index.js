/* @flow */
import React, {PropTypes as T} from 'react';

import style from './style.css';

const Badge = (
  {children, title}/*: {children?: string | number, title?: string} */
) => (
  <span className={style.badge} title={title || children}>
    {children}
  </span>
);
Badge.propTypes = {
  children: T.oneOfType([T.string, T.number]).isRequired,
  title: T.string,
};

export default Badge;
