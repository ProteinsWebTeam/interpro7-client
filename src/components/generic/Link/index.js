// @flow
import React, { PureComponent } from 'react';
import T from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import descriptionToDescription from 'utils/processDescription/descriptionToDescription';

import { goToCustomLocation, closeEverything } from 'actions/creators';
import { customLocationSelector } from 'reducers/custom-location';

import generateHref from './utils/generate-href';
import generateRel from './utils/generate-rel';
import generateClassName from './utils/generate-classname';

// $FlowFixMe
import cssBinder from 'styles/cssBinder';
import buttonCSS from 'components/SimpleCommonComponents/Button/style.css';
const css = cssBinder(buttonCSS);

import { DEV } from 'config';

const happenedWithModifierKey = (event) =>
  !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
const happenedWithLeftClick = (event) => event.button === 0;

const getNextLocation = (customLocation, to) =>
  typeof to === 'function' ? to(customLocation) : to;

/*:: type Props = {
  onClick?: function,
  onNewLocation?: function,
  customLocation: {
    description: Object,
    search: Object,
    hash: string,
  },
  exact?: boolean,
  children: any,
  rel?: string,
  href?: string,
  closeEverything: function,
  goToCustomLocation: function,
  target?: string,
  to?: function | {
    description: Object,
    search?: Object,
    hash?: string,
  },
  style?: ?Object,
  disabled?: boolean,
  className?: string,
  activeClass?: function | string,
  withReferrer?: boolean,
  buttonType?: string,
}; */

export class _Link extends PureComponent /*:: <Props> */ {
  static propTypes = {
    id: T.string,
    onClick: T.func,
    onNewLocation: T.func,
    customLocation: T.shape({
      description: T.object.isRequired,
      search: T.object.isRequired,
      hash: T.string.isRequired,
    }).isRequired,
    exact: T.bool,
    children: T.any,
    rel: T.string,
    href: T.string,
    closeEverything: T.func.isRequired,
    goToCustomLocation: T.func.isRequired,
    target: T.string,
    to: T.oneOfType([
      T.shape({
        description: T.object.isRequired,
        search: T.object,
        hash: T.string,
      }),
      T.func,
    ]),
    disabled: T.bool,
    className: T.string,
    activeClass: T.oneOfType([T.string, T.func]),
    withReferrer: T.bool,
    download: T.string,
    buttonType: T.string,
  };

  handleClick = (event) => {
    const {
      disabled,
      onClick,
      onNewLocation,
      target,
      closeEverything,
      goToCustomLocation,
      to,
      href,
      customLocation,
    } = this.props;
    // pass it on to an externally defined handler
    if (disabled) {
      event.preventDefault();
      return;
    }

    if (onClick) onClick(event);
    if (event.defaultPrevented) return;
    closeEverything();
    if (!to && href) return;
    // conditions to ignore handling
    if (!happenedWithLeftClick(event)) return;
    if (happenedWithModifierKey(event)) return;
    if (target) return;
    // OK, now we can handle it
    event.preventDefault();
    if (this.props.disabled) return;
    const nextLocation = getNextLocation(customLocation, to);
    goToCustomLocation(nextLocation);
    if (onNewLocation) onNewLocation(nextLocation);
  };

  render() {
    const {
      // unused (to prevent passing down)
      onClick,
      closeEverything,
      goToCustomLocation,
      // used or changed
      exact,
      customLocation,
      activeClass,
      className,
      to,
      disabled,
      href,
      children,
      rel,
      target,
      withReferrer,
      buttonType,
      // passed down
      ...props
    } = this.props;
    let _disabled = disabled;
    let _children = children;
    let _href;
    let activeClassName;
    try {
      const nextCustomLocation = getNextLocation(customLocation, to) || {};
      nextCustomLocation.description = descriptionToDescription(
        nextCustomLocation.description,
      );
      _href = generateHref(nextCustomLocation, href);
      activeClassName = generateClassName(
        activeClass,
        customLocation,
        href,
        _href,
        exact,
      );
    } catch (error) {
      console.error(error);
      _disabled = true;
      if (DEV) {
        props.style = {
          ...(props.style || {}),
          border: '5px dashed red',
          color: 'darkred',
        };
        _children = (
          <>
            ⚠️
            {children}
            ⚠️
          </>
        );
      }
    }
    if (_disabled) {
      props.style = {
        ...(props.style || {}),
        // userSelect: 'none',
        pointerEvents: 'none',
        cursor: 'not-allowed',
        opacity: 0.3,
      };
    }
    const _rel = generateRel(rel, target, href, withReferrer);
    const buttonClassName = buttonType
      ? ['vf-button', `vf-button--${buttonType}`, 'vf-button--sm']
      : [];
    return (
      <a
        {...props}
        href={_href}
        rel={_rel}
        target={target}
        className={css(className, activeClassName, ...buttonClassName) || null}
        onClick={this.handleClick}
        style={{
          // Corrections for interpro-new... to remove when that css is finally removed
          color: buttonType
            ? 'var(--vf-button-text-color, #ffffff)'
            : undefined,
          borderBottomColor: buttonType
            ? 'var(--vf-button-border-color, #ffffff)'
            : undefined,
        }}
      >
        {_children}
      </a>
    );
  }
}
_Link.displayName = 'Link';
const mapStateToProps = createSelector(
  customLocationSelector,
  (customLocation) => ({ customLocation }),
);

export default connect(mapStateToProps, {
  closeEverything,
  goToCustomLocation,
})(_Link);
