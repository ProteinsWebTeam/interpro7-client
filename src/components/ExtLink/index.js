/* eslint no-magic-numbers: [1, {ignore: [3]}] */
import React from 'react';
import Link from 'components/generic/Link';
import T from 'prop-types';

const types = {
  id: T.oneOfType([T.string, T.number]).isRequired,
  children: T.node,
  className: T.string,
  target: T.string,
};

export const BaseLink = ({
  id,
  pattern,
  href,
  newTo,
  to,
  rel,
  className,
  children,
  activeClass,
  ...rest
}) => {
  const props = {
    href: href || newTo || to || (pattern || '').replace('{id}', id),
  };
  if (className) props.className = className;
  if (rel) {
    props.rel = rel.includes('noopener') ? rel : `${rel} noopener`;
    props.rel = rel.includes('noreferrer') ? rel : `${rel} noreferrer`;
  } else {
    props.rel = 'noopener noreferrer';
  }
  return (
    <Link {...rest} {...props}>
      {children}
    </Link>
  );
};
BaseLink.propTypes = {
  id: T.oneOfType([T.string, T.number]),
  pattern: T.string,
  href: T.string,
  rel: T.string,
  to: T.string,
  newTo: T.object,
  target: T.string,
  className: T.string,
  activeClass: T.oneOfType([T.string, T.func]),
  children: T.node.isRequired,
};
BaseLink.displayName = 'BaseLink';

const patternLinkWrapper = pattern => {
  const Wrapped = ({ id, target, children, ...props }) => (
    <BaseLink id={id} target={target || '_blank'} pattern={pattern} {...props}>
      {children || id}
    </BaseLink>
  );
  Wrapped.propTypes = types;
  return Wrapped;
};

export const TaxLink = ({ id, target, children, ...props }) => (
  <BaseLink
    id={id}
    target={target || '_blank'}
    pattern="https://www.ebi.ac.uk/ena/data/view/Taxon:{id}"
    {...props}
  >
    {children || `TaxID ${id}`}
  </BaseLink>
);
TaxLink.propTypes = {
  ...types,
  id: T.number.isRequired,
};
TaxLink.displayName = 'TaxLink';

export const ProteomeLink = ({ id, target, children, ...props }) => (
  <BaseLink
    id={id}
    target={target || '_blank'}
    pattern="http://www.uniprot.org/proteomes/{id}"
    {...props}
  >
    {children || `ProteomeID ${id}`}
  </BaseLink>
);
ProteomeLink.propTypes = {
  ...types,
  id: T.string.isRequired,
};
ProteomeLink.displayName = 'ProteomeLink';

export const PMCLink = ({ id, target, children, ...props }) => (
  <BaseLink
    id={id}
    target={target || '_blank'}
    pattern="https://europepmc.org/abstract/MED/{id}"
    {...props}
  >
    {children || `PUB${id}`}
  </BaseLink>
);
PMCLink.propTypes = {
  ...types,
  id: T.string.isRequired,
};
PMCLink.displayName = 'PMCLink';

export const DOILink = patternLinkWrapper('{id}');
DOILink.displayName = 'DOILink';

export const GoLink = ({ id, target, className, children, ...props }) => {
  const pattern = 'https://www.ebi.ac.uk/QuickGO/GTerm?id={id}';
  return (
    <BaseLink
      id={id.replace('_', ':')}
      target={target || '_blank'}
      pattern={pattern}
      {...(className ? { className } : {})}
      {...props}
    >
      {children || id.replace('_', ':')}
    </BaseLink>
  );
};
GoLink.propTypes = types;
GoLink.displayName = 'GoLink';

export const PDBeLink = patternLinkWrapper(
  'https://www.ebi.ac.uk/pdbe/entry/pdb/{id}'
);
PDBeLink.displayName = 'PDBeLink';

export const PDBe3DLink = patternLinkWrapper(
  'https://www.ebi.ac.uk/pdbe/entry/view3D/{id}/' +
    '?view=entry_index&viewer=jmol&controls=codename_hero'
);
PDBe3DLink.displayName = 'PDBe3DLink';

export const UniProtLink = patternLinkWrapper(
  'http://www.uniprot.org/uniprot/{id}'
);
UniProtLink.displayName = 'UniProtLink';

const ExtLink = ({ id, children, ...props }) => {
  switch (true) {
    case id.startsWith('PUB'):
      return (
        <PMCLink id={id.slice(3)} {...props}>
          {children}
        </PMCLink>
      );
    default:
      throw Error('Not a supported reference link');
  }
};
ExtLink.propTypes = types;
ExtLink.displayName = 'ExtLink';

export default ExtLink;
