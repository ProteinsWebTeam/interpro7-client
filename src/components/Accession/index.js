import React, { PureComponent } from 'react';
import T from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import { UniProtLink, ProteomeLink } from 'components/ExtLink';

const Default = ({ children }) => <span>{children}</span>;
Default.propTypes = {
  children: T.oneOfType([T.string, T.number]).isRequired,
};

/*:: type Props = {
  accession: string | number,
  db: string,
}; */

export class Accession extends PureComponent /*:: <Props> */ {
  static propTypes = {
    accession: T.oneOfType([T.string, T.number]).isRequired,
    title: T.string,
    db: T.string,
  };

  render() {
    const { accession, id, db, title } = this.props;
    return (
      <div>
        {title || 'Accession'}: {accession}
      </div>
    );
  }
}

const mapStateToProps = createSelector(
  state =>
    state.customLocation.description[state.customLocation.description.main.key]
      .db,
  db => ({ db }),
);

export default connect(mapStateToProps)(Accession);
