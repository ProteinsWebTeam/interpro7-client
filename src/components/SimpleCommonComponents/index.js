// @flow
import React, {PropTypes as T, createElement} from 'react';
import {Link} from 'react-router-dom';

import {TaxLink, PDBeLink, UniProtLink} from 'components/ExtLink';

import {buildLink} from 'utils/url';

export const Name = (
  {name: {name, short}, accession}
  /*: {name: {name: string, short: ?string}, accession: string} */
) => (
  <div>
    <h3>{name} ({accession})</h3>
    {short && <p style={{color: 'gray'}}>Short name: {short}</p>}
  </div>
);
Name.propTypes = {
  name: T.shape({
    short: T.string,
    name: T.string.isRequired,
  }).isRequired,
  accession: T.string.isRequired,
};

const extOriginDBLookup = {
  pdb: {
    nicerName: 'PDBe',
    component: PDBeLink,
  },
  swissprot: {
    nicerName: 'SwissProt',
    component: UniProtLink,
  },
  trembl: {
    nicerName: 'TrEMBL',
    component: UniProtLink,
  },
};

export const ExtOriginDB = (
  {source, accession}/*: {source: string, accession: string | number} */
) => {
  const desc = extOriginDBLookup[source.toLowerCase()];
  if (!desc) return null;
  return createElement(
    desc.component,
    {id: accession},
    [`(${desc.nicerName || source} external link)`]
  );
};
ExtOriginDB.propTypes = {
  source: T.string.isRequired,
  accession: T.oneOfType([T.string, T.number]).isRequired,
};

export const OriginDB = (
  {source, pathname, accession}
  /*: {source: string, pathname: string, accession: string | number} */
) => (
  <p>
    Source DB:&nbsp;
    <Link to={buildLink(pathname, pathname.split('/')[1], source)}>
      {source}
    </Link>&nbsp;
    <ExtOriginDB source={source} accession={accession} />
  </p>
);
OriginDB.propTypes = {
  source: T.string.isRequired,
  pathname: T.string.isRequired,
  accession: T.oneOfType([T.string, T.number]).isRequired,
};

export const SourceOrganism = (
  {taxid, name}/*: {taxid: number, name: string} */
) => (
  <p>
    Source Organism: <TaxLink id={taxid}>{name} ({taxid})</TaxLink>
  </p>
);
SourceOrganism.propTypes = {
  taxid: T.number.isRequired,
  name: T.string.isRequired,
};
