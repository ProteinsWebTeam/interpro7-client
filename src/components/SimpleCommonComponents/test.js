/* eslint-env node */
/* eslint-env mocha */
/* eslint no-magic-numbers: [1, {ignore: [12345]}]*/
import 'babel-polyfill';

import React, {createElement} from 'react';
import {createRenderer} from 'react-dom/test-utils';
import chai, {expect} from 'chai';
import jsxChai from 'jsx-chai';

import Link from 'components/generic/Link';

import {Name, ExtOriginDB, OriginDB, SourceOrganism} from '.';
import {TaxLink, PDBeLink, UniProtLink} from 'components/ExtLink';

chai.use(jsxChai);
const renderer = createRenderer();

describe('Simple Common Components', () => {
  describe('<Name />', () => {
    it('should render name information', () => {
      renderer.render(
        <Name name={{name: 'name', short: 'short'}} accession="accession" />
      );
      expect(renderer.getRenderOutput()).to.deep.equal(
        <div>
          <h3>name (accession)</h3>
          <p style={{color: 'gray'}}>Short name: short</p>
        </div>
      );
      renderer.render(
        <Name name={{name: 'name'}} accession="accession" />
      );
      expect(renderer.getRenderOutput()).to.deep.equal(
        <div>
          <h3>name (accession)</h3>
        </div>
      );
    });
  });

  describe('<ExtOriginDB />', () => {
    it('should render external origin database link', () => {
      const sourceComponentTuples = [
        ['pdb', PDBeLink, 'PDBe'],
        ['PDB', PDBeLink, 'PDBe'],
        ['swissprot', UniProtLink, 'SwissProt'],
        ['SwissProt', UniProtLink, 'SwissProt'],
        ['trembl', UniProtLink, 'TrEMBL'],
        ['TrEMBL', UniProtLink, 'TrEMBL'],
      ];
      for (const accession of ['accession', 12345]) {
        for (const [source, component, nicerName] of sourceComponentTuples) {
          renderer.render(
            <ExtOriginDB source={source} accession={accession} />
          );
          expect(renderer.getRenderOutput()).to.deep.equal(
            createElement(
              component,
              {id: accession},
              [`(${nicerName} external link)`]
            )
          );
        }
      }
    });

    it('should render nothing if not an external database', () => {
      for (const notExternalDB of ['InterPro', 'interpro', 'randomString']) {
        renderer.render(<ExtOriginDB source={notExternalDB} accession={''} />);
        expect(renderer.getRenderOutput()).to.deep.equal(null);
      }
    });
  });

  describe.skip('<OriginDB />', () => {
    it('should render origin database information', () => {
      const fixtureTuples = [
        ['protein', 'trembl', 'A0JUS0'],
        ['entry', 'interpro', 'IPR000001'],
        ['structure', 'pdb', '9xim'],
      ];
      for (const [type, db, accession] of fixtureTuples) {
        renderer.render(
          <OriginDB
            source={db}
            pathname={`/${type}/${db}/${accession}/`}
            accession={accession}
          />
        );
        expect(renderer.getRenderOutput()).to.deep.equal(
          <p>
            Source DB:{' '}
            <Link
              newTo={{description: {mainType: type, mainDB: db}}}
            >
              {db}
            </Link> <ExtOriginDB source={db} accession={accession} />
          </p>
        );
      }
    });
  });

  describe('<SourceOrganism />', () => {
    it('should render source organism information', () => {
      renderer.render(
        <SourceOrganism taxid={12345} name="organism name" />
      );
      expect(renderer.getRenderOutput()).to.deep.equal(
        <p>
          Source Organism:
          <TaxLink id={12345}>organism name (12345)</TaxLink>
        </p>
      );
    });
  });
});
