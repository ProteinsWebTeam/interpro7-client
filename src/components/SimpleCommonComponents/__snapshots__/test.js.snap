// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 1`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 2`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 3`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 4`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 5`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 6`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 7`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 8`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 9`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 10`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 11`] = `null`;

exports[`Simple Common Components <ExtOriginDB /> should render external origin database link 12`] = `null`;

exports[`Simple Common Components <Name /> should render name information 1`] = `
<div>
  <h3>
    name
     (
    accession
    )
  </h3>
  <p
    style={
      Object {
        "color": "gray",
      }
    }
  >
    Short name: 
    short
  </p>
</div>
`;

exports[`Simple Common Components <Name /> should render name information 2`] = `
<div>
  <h3>
    name
     (
    accession
    )
  </h3>
</div>
`;

exports[`Simple Common Components <SourceOrganism /> should render source organism information 1`] = `
<p>
  Source Organism:
   
  <TaxLink
    id={12345}
  >
    organism name
     (
    12345
    )
  </TaxLink>
</p>
`;
